### PR TITLE
adding ctor with optional comments indices args

### PR DESCRIPTION
**Pull request by toepkerd**
### Description
Fixing a breaking change to Security Analytics Plugin caused by https://github.com/opensearch-project/common-utils/pull/663
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt [hunk 1]
```diff
@@ -89,6 +86,29 @@ data class DataSources(
         findingsEnabled = sin.readOptionalBoolean()
     )
 
+    @Throws(IOException::class)
```
Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt:89]: why are we expecting IoException?

Comment (toepkerd) [reply src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt:89]: removing

**Review by eirsep — APPROVED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt [hunk 2]
```diff
@@ -89,6 +86,28 @@ data class DataSources(
         findingsEnabled = sin.readOptionalBoolean()
     )
 
+    constructor(
```
Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt:89]: https://github.com/opensearch-project/common-utils/pull/674/files#diff-97cf101698aba14b5844f08b1d5be6fcf2752671b9d7510d1a566efbeec20793R83

comments index should be using readOptionalString(), writeOptionalString, null check in parsing etc. 
plz incorporate

**Review by eirsep — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt (hunk 2 above)
Comment (toepkerd) [reply src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt:89]: updating readOptional and writeOptional, parse adds an empty string placeholder for any comments that don't exist in the XContent, so leaving that be

**Review by eirsep — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**